test/e2e/common_suite_test.go

```diff
@@ -672,3 +727,50 @@ func doTestCreatePeerPodWithPVCAndCSIWrapper(t *testing.T, assert CloudAssert, m
 	}
 	newTestCase(t, "PeerPodWithPVCAndCSIWrapper", assert, "PVC is created and mounted as expected").withPod(pod).withPVC(myPVC).withTestCommands(testCommands).run()
 }
+
+func doTestCreatePeerPodWithAuthenticatedImage(t *testing.T, assert CloudAssert) {
+	namespace := envconf.RandomName("default", 7)
+	podName := "authenticated-image-valid-pod"
+	secretName := "auth-json-secret"
+	authfile, err := os.ReadFile("../../install/overlays/ibmcloud/auth.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectedAuthStatus := "Completed"
+	secretData := map[string][]byte{v1.DockerConfigJsonKey: authfile}
+	secret := newSecret(namespace, secretName, secretData, v1.SecretTypeDockerConfigJson)
+	imageName := os.Getenv("REGISTRY_IMAGE")
+	pod := newPod(namespace, podName, podName, imageName, withRestartPolicy(v1.RestartPolicyNever), withImagePullSecrets(secretName))
+	newTestCase(t, "AuthImagePeerPod", assert, "Peer pod with Authenticated Image has been created").withSecret(secret).withPod(pod).withAuthenticatedImage().withAuthImageStatus(expectedAuthStatus).withCustomPodState(v1.PodSucceeded).run()
+}
+
+func doTestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials(t *testing.T, assert CloudAssert) {
+	namespace := envconf.RandomName("default", 7)
+	podName := "authenticated-image-invalid-pod"
+	secretName := "auth-json-secret"
+	data := map[string]interface{}{
+		"auths": map[string]interface{}{
+			"quay.io": map[string]interface{}{
+				"auth": "aW52YWxpZHVzZXJuYW1lOmludmFsaWRwYXNzd29yZAo=",
+			},
+		},
+	}
+	jsondata, err := json.MarshalIndent(data, "", " ")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := ioutil.WriteFile("../../install/overlays/ibmcloud/auth.json", jsondata, 0644); err != nil {
+		t.Fatal(err)
+	}
+	authfile, err := os.ReadFile("../../install/overlays/ibmcloud/auth.json")
```

Comment:
Can we not just use `jsondata` directly in the `map[string][]byte{v1.DockerConfigJsonKey: authfile}` ? Rather than writing and reading the same file from disk?